src/openlcb/IfImpl.hxx

```diff
@@ -266,6 +266,78 @@ private:
     If::VNodeMap::Iterator it_;
 #endif
 };
+
+/// Message handler that is registered as a fallback handler in the interface's
+/// message dispatcher. This means that all incoming messages that were not
+/// matching the MTI / mask of any existing instantiated handler will end up
+/// here.
+///
+/// THe standard requires that when an addressed message is not handled by a
+/// node (e.g. because it does not know about the MTI at all), then an Optional
+/// Interaction Rejected reply be sent to the originator. This flow generates
+/// that OIR reply.
+class UnhandledAddressedMessageHandler : public IncomingMessageStateFlow
+{
+public:
+    UnhandledAddressedMessageHandler(If *service)
+        : IncomingMessageStateFlow(service)
+    {
+        iface()->dispatcher()->register_fallback_handler(this);
+    }
+
+    /// Handler callback for incoming messages.
+    Action entry() override
+    {
+        if (!message()->data()->dstNode)
+        {
+            // Destination is not a local node.
+            return release_and_exit();
+        }
+        return allocate_and_call(
+            iface()->addressed_message_write_flow(), STATE(fill_oir));
+    }
+
+    Action fill_oir()
```

Comment:
Could we add Doxygen header blocks to this method, and the others that are undocumented in this file?

Reply:
done